services/core-web/src/components/noticeOfWork/applications/process/ProcessPermit.js

```diff
@@ -349,44 +362,202 @@ export class ProcessPermit extends Component {
+            !this.props.progress[progressStatus.application_progress_status_code].end_date)
+      )
+      .forEach((progressStatus) =>
+        validationMessages.push({
+          message: `${progressStatus.description} must be completed.`,
+          route: ProgressRouteFor(
+            progressStatus.application_progress_status_code,
+            this.props.noticeOfWork?.now_application_guid
+          ),
+        })
+      );
+
+    if (this.props.progress.CON?.start_date && !this.props.progress.CON?.end_date) {
+      validationMessages.push({
+        message: "Consultation must be completed.",
+        route: ProgressRouteFor("CON", this.props.noticeOfWork?.now_application_guid),
+      });
     }
+
+    return validationMessages;
+  };
+
+  getValidationWarnings = () => {
+    const validationMessages = [];
+
+    // Mine Emergency Resposne Plan
     if (
       !this.props.noticeOfWork.documents ||
       !this.props.noticeOfWork.documents.some(
         (doc) => doc.now_application_document_type_code === "MRP" && doc.is_final_package
       )
     ) {
       validationMessages.push({
-        message: `The final application package requires a Mine Emergency Response Plan.`,
+        message: `The final application package is missing a Mine Emergency Response Plan. You can Edit the Final Application Package under Administrative.`,
+        route: route.NOTICE_OF_WORK_APPLICATION.dynamicRoute(
+          this.props.noticeOfWork.now_application_guid,
+          "administrative"
+        ),
       });
     }
-    this.props.progressStatusCodes
-      .filter(
-        (progressStatus) =>
-          !this.props.progress[progressStatus.application_progress_status_code] ||
-          !this.props.progress[progressStatus.application_progress_status_code].end_date
+
+    // Archeological Chance Find Procedure
+    if (
+      !this.props.noticeOfWork.documents ||
+      !this.props.noticeOfWork.documents.some(
+        (doc) => doc.now_application_document_type_code === "ACP" && doc.is_final_package
       )
-      .forEach((progressStatus) =>
-        validationMessages.push({ message: `${progressStatus.description} must be completed.` })
-      );
+    ) {
+      validationMessages.push({
+        message: `The final application package is missing an Archaeological Chance Find Procedure. You can Edit the Final Application Package under Administrative.`,
+        route: route.NOTICE_OF_WORK_APPLICATION.dynamicRoute(
+          this.props.noticeOfWork.now_application_guid,
+          "administrative"
+        ),
+      });
+    }
+
+    // Permittee
+    if (
+      this.props.noticeOfWork.contacts &&
+      this.props.noticeOfWork.contacts.some(
+        (contact) => contact.mine_party_appt_type_code === "PMT"
+      )
+    ) {
+      const permittee = this.props.noticeOfWork.contacts.filter(
+        (contact) => contact.mine_party_appt_type_code === "PMT"
+      )[0];
```

Comment:
Index out of range error if no permittee has been set?

Reply:
no, there's a .some() conditional above